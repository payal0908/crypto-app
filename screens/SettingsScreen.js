import React, {useState, useEffect} from 'react';
import { View, Text, StyleSheet, Switch } from 'react-native';
import { useTheme } from '@react-navigation/native';
import { EventRegister } from 'react-native-event-listeners';
import AsyncStorage from '@react-native-async-storage/async-storage';

const SettingsScreen = ({navigation}) => {

  const [darkMode, setDarkMode] = useState(false)
  const { colors } = useTheme();

  useEffect(() => {
    const getDarkMode = async () => {
      try {
        let getMode = await AsyncStorage.getItem('darkMode');
        setDarkMode(JSON.parse(getMode))
      } catch(err) {
        console.log(err);
      }
    }
    getDarkMode();
  })

  return (
    <View style={[styles.container, {backgroundColor: colors}]}>
      <View style={[styles.data, {backgroundColor: colors.settingsBackground, shadowColor: colors.settingsShadow}]}>
        <Text style={[styles.headerText, {color: colors.text}]}>Dark Mode</Text>
        <Switch
          value={darkMode}
          onValueChange={val => {
            setDarkMode(val);
            EventRegister.emit('changeThemeEvent', val);
          }}/>
        </View>
    </View>
  );
}

export default SettingsScreen;

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: "center",
      justifyContent: "center",
      backgroundColor: "white"
    },
    headerText: {
      fontSize: 25,
      fontWeight: '500',
      marginBottom: 15
    },
    data: {
      alignItems: 'center',
      justifyContent: 'center',
      padding: 16,
      height: 180,
      borderRadius: 15,
      backgroundColor: '#f6fae6',
      shadowColor: "#7a72d0",
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.51,
      shadowRadius: 4.11,
      elevation: 14,
    }
});
