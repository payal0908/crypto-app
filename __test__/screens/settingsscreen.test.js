import React from 'react';
import {create} from 'react-test-renderer';

import SettingsScreen from '../../screens/SettingsScreen';

const tree = create(<SettingsScreen />)

test('snapshot', () => {
  expect(tree).toMatchSnapshot()
});
