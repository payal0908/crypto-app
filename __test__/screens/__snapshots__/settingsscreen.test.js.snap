// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": Object {
          "background": "rgb(242, 242, 242)",
          "border": "rgb(216, 216, 216)",
          "card": "rgb(255, 255, 255)",
          "notification": "rgb(255, 59, 48)",
          "primary": "rgb(0, 122, 255)",
          "text": "rgb(28, 28, 30)",
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6fae6",
          "borderRadius": 15,
          "elevation": 14,
          "height": 180,
          "justifyContent": "center",
          "padding": 16,
          "shadowColor": "#7a72d0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.51,
          "shadowRadius": 4.11,
        },
        Object {
          "backgroundColor": undefined,
          "shadowColor": undefined,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 25,
            "fontWeight": "500",
            "marginBottom": 15,
          },
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      Dark Mode
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={false}
    />
  </View>
</View>
`;
